src/Platforms/Echo.Platforms.AsmResolver/Emulation/Values/DefaultValueFactory.cs

```diff
@@ -175,13 +175,13 @@ private IConcreteValue CreateUnknown(TypeSignature type)
         }
 
         /// <inheritdoc />
-        public MemoryPointerValue AllocateMemory(int size, bool initialize)
+        public IMemoryAccessValue AllocateMemory(int size, bool initialize)
         {
             var memory = new Memory<byte>(new byte[size]);
```

Comment:
If we do decide to use array pooling in MemoryBlockValue, these allocations can also be avoided with stackalloc's.

Reply:
How could we leverage stackalloc here, if the allocated memory has to outlive the scope of `AllocateMemory` ?

Reply:
If you use array pooling inside, MemoryPointerValue, it could take in spans and copy their content to the rented array. The rented array would of course be owned my the MemoryPointerValue instance.

Reply:
Ah you want to give the memory block value a constructor taking an `int size` and `bool initialize`, and rent there? Yea that could work.